### ISSUE TITLE
Wrong colors after resize

### ISSUE THREAD

**Issue by pftbest** [nfnt/resize#31]
i wrote this simple program and it works but resulting image has color artifacts in it:
[Source image](https://googledrive.com/host/0B4ouTzM6vLORflJYVGZHUzk2VTZpRW1KUzh5VEhOWXNGVlBlbjBRYlZzcEJBTzNkWFpneVE/1.jpg)
[Result image](https://googledrive.com/host/0B4ouTzM6vLORflJYVGZHUzk2VTZpRW1KUzh5VEhOWXNGVlBlbjBRYlZzcEJBTzNkWFpneVE/1.png)
[Expected image](https://googledrive.com/host/0B4ouTzM6vLORflJYVGZHUzk2VTZpRW1KUzh5VEhOWXNGVlBlbjBRYlZzcEJBTzNkWFpneVE/1a.png)

code:
```Go
package main

import (
	"github.com/nfnt/resize"
	"image"
	_ "image/jpeg"
	"image/png"
	"os"
)

func main() {
	reader, _ := os.Open("test/1.jpg")
	defer reader.Close()
	m, _, _ := image.Decode(reader)
	m = resize.Resize(32, 32, m, resize.Bicubic)
	file, _ := os.OpenFile("test/1.png", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
	png.Encode(file, m)
	defer file.Close()
}
```

**Comment by pftbest** [comment]
curiously enough this code works fine:
```Go
package main

import (
	"github.com/disintegration/imaging"
	"github.com/nfnt/resize"
	"os"
)

func main() {
	reader, _ := os.Open("test/1.jpg")
	defer reader.Close()
	m, _ := imaging.Decode(reader)
	m = resize.Resize(32, 32, m, resize.Bicubic)
	file, _ := os.OpenFile("test/1.png", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
	imaging.Encode(file, m, imaging.PNG)
	defer file.Close()
}
```
but `disintegration/imaging` library has its own resize function, so there is no point in using them both together.

**Comment by nfnt** [comment]
I can see the artifacts in the resulting image but judging from your code it seems that the resize package is not responsibly for that. The only thing that's changed is the call to `imaging.Encode` which wraps `png.Encode`, i.e. both samples /should/ do exactly the same. How did you create the expected image?

**Comment by pftbest** [comment]
expected image was created using resize function from imagine library using the code similar to my second example.
i think the reason for such strange behaviour is that im loading image using `imagine.Decode`. it always converts image to NRGBA format and sets image boundaries to start from (0,0).

P.S. have you tried to run my code? can you reproduce the issue?

**Comment by sharpner** [comment]
I had the same problem: Upscaling this image resulted in an very white and strange coloring. 

![testfile](https://cloud.githubusercontent.com/assets/574362/7743940/9c3608e0-ff9f-11e4-84f2-b7b6f3e109d0.png)

**Comment by nfnt** [comment]
Fixed with commit dc93e1b. This would only happen with JPEG images, as these use chroma subsampling.
